appside/frames/body.js

```diff
@@ -365,39 +378,50 @@ class BodyMapColorFrame extends Frame {
-                $(body_link).text(part.charAt(0).toUpperCase() + part.slice(1));
-                body_link.onclick = function () {   // clicked on body part
-                    this.bodypart = part;
-                    this.render_left_col(frame);
-                    this.render_right_col(frame);
-                }.bind(this);
-                frame.right.appendChild(body_link);
-            }
-        } else {    // body part is selected
-            let question = document.createElement('p');
-            var string = this.bodypart + ' in ' + this.emotion;
-            $(question).text(this.question.replace('{}', string));
-            frame.right.appendChild(question);
-
-            for (let choice of this.qualifiers) {
-                let radio = document.createElement('input');
-                $(radio).attr('type', 'radio');
-                $(radio).attr('name', 'emotion');
-                $(radio).attr('id', choice);
-
-                let label = document.createElement('label');
-                $(label).attr('for', choice);
-                $(label).text(choice);
-                frame.right.appendChild(radio);
-                frame.right.appendChild(label);
-                frame.right.appendChild(document.createElement('br'));
-            }
+        frame.right.innerHTML = '';
+        let question = document.createElement('p');
+        var string = this.bodypart + ' in ' + this.emotion;
+        $(question).text(this.question.replace('{}', string));
+        frame.right.appendChild(question);
+
+        for (let choice of this.qualifiers) {
+            let radio = document.createElement('input');
+            $(radio).attr('type', 'radio');
+            $(radio).attr('name', this.emotion + '_' + this.bodypart);
+            $(radio).attr('id', choice);
+
+            let label = document.createElement('label');
+            $(label).attr('for', choice);
+            $(label).text(choice);
+            frame.right.appendChild(radio);
+            frame.right.appendChild(label);
+            frame.right.appendChild(document.createElement('br'));
         }
+
+        // next will be implemented in navigator?
+        let next = document.createElement('button');
+        $(next).attr('class', 'bodymap_color_button');
+        $(next).text('Next');
+        $(next).click(function() {
+            var choices = document.getElementsByTagName('input');;
+            for (let each of choices) {
+                if (each.checked) { this.answer = each.id; } // only one checked
+            }
+            this.user_input = new Map();
+            this.user_input.set('emotion', this.emotion);
+            this.user_input.set('bodypart', this.bodypart);
+            this.user_input.set('answer', this.answer);
+        }.bind(this));
+        frame.right.appendChild(next);
+
+    }
+
+    /**
+     * Returns data of user input
+     * including emotion, bodypart, and answer (qualifier)
+     */
+    userInput() {
```

Comment:
this should be called `get_user_input`

(same reason as the other one)

Reply:
Changed method function accordingly in BodyMapFrame and BodyMapColorFrame